### ISSUE TITLE
[3.0.x]Exception using Wc_order and false when I use wc_get_order

### ISSUE THREAD

**Issue by isidro1984** [woocommerce/woocommerce#15616]
### EXPLANATION OF THE ISSUE

I am trying to get orders in a function on my template using the file functions.php (then I will build a plugin when it works). The problem is after updating to Woocommerce 3.0 I am not able to get the orders anyway. It seems that the orders don't exist but they are in the DB and in the backend.  

For example, there is an order with the post_id = 10 made with the Woocommerce Cart. 

I'm trying to get it with: 
`$order = wc_get_order(10);`

Then I always receive false.

I tried instancing the class sending the ID as a parameter:

`new WC_Order( 10 );`

But if we use this option that I used in the past, I receive the following exception:

Uncaught exception 'Exception' with message 'Invalid order.' in C:\xampp\htdocs\bodas_wc\wp-content\plugins\woocommerce\includes\data-stores\abstract-wc-order-data-store-cpt.php:86 Stack trace: #0 C:\xampp\htdocs\bodas_wc\wp-content\plugins\woocommerce\includes\class-wc-data-store.php(143): Abstract_WC_Order_Data_Store_CPT->read(Object(WC_Order)) #1 C:\xampp\htdocs\bodas_wc\wp-content\plugins\woocommerce\includes\abstracts\abstract-wc-order.php(109): WC_Data_Store->read(Object(WC_Order)) #2 C:\xampp\htdocs\bodas_wc\wp-content\themes\twentyseventeen\functions.php(585): WC_Abstract_Order->__construct('5941128984246') #3 [internal function]: getWC_order_details('') #4 C:\xampp\htdocs\bodas_wc\wp-includes\class-wp-hook.php(298): call_user_func_array('getWC_order_det...', Array) #5 C:\xampp\htdocs\bodas_wc\wp-includes\class-wp-hook.php(323): WP_Hook->apply_filters('', Array) #6 C:\xampp\htdocs\bodas_wc\wp-includes\plugin.php(453): WP_Hook->do_action(Array) #7 C:\xampp\htdocs\bodas_wc\wp-content\plugins\woocommerce in C:\xampp\htdocs\bodas_wc\wp-content\plugins\woocommerce\includes\data-stores\abstract-wc-order-data-store-cpt.php on line 86

Most rare is the next thing, If I try to get the orders with the default function get_posts(), it is not able to find anything with the post_type = 'shop_orders'; I tried with 'post_type' = 'products' and in this case it works. 

Its very weird because its the same code that I have in other projects and it works if the woocommerce version is < 3.0

Tested in Wamp and Cloudways server with this configuration:

Wordpress 4.8
Woocommerce - 3.0.0 and 3.0.8 
PHP 5.6
Apache 2.4

Wordpress 

### STEPS TO REPRODUCE THE ISSUE

Install Wordpress - Woocommerce and add this function to the functions.php template:

```
function getWC_order_details($order_id) {    
	$order_id = 10;
	//$order = new WC_Order(10); optionally I tried this
	$order = wc_get_order(10);
	//$order->get_items(); This is the next step
	var_dump($order);
}

add_action('woocommerce_init', 'getWC_order_details');
//I changed this hook aswell but nothing happens. 
```

### SYSTEM STATUS

<details>
```
Grab the system status report from WooCommerce > System Status and paste it here between the `details` tags.
```
</details>

**Comment by mikejolley** [comment]
`wc_get_order` is the correct method to use. If you get false/invalid order exception it's indeed that.  Either the order does not exist, or has the wrong type. `shop_order` is the type FYI.

**Comment by renatonagliati** [comment]
Hi,

I'm having the same issue. Could you check, please?

## Explanation ##
Basically I'm getting all "processing orders created after 2017-07-01" and passing through a loop. I'm having two issues on this process:

1. The WC_Order_Query returns the orders only if the parm 'return' is defined - without this parm, I'm getting an empty array;

```
$query = new WC_Order_Query( array(
	'limit' => -1,
	'type' => 'shop_order',
	'status' => 'processing',
	'date_created' => '>2017-07-01',
	'return' => 'ids', //if I comment this line, the result will be empty
) );
$processing_orders = $query->get_orders();
```

2. With the $processing_orders [on my DB I have only 1 order; with ID 213], the wc_get_order is returning false:

```
if(!empty($processing_orders)){

	foreach($processing_orders as $order_id){
		$order = wc_get_order( intval($order_id) );
		var_dump( $order ); //is returning bool(false)
	}
	
}
```


## Print Screen ##
Query: SELECT * FROM `wp_posts` WHERE post_type = 'shop_order'
Result: 
![image](https://user-images.githubusercontent.com/7807925/27932408-0c75d714-6274-11e7-96af-53160045443b.png)

WooCommerce > Orders
![image](https://user-images.githubusercontent.com/7807925/27932459-38467b14-6274-11e7-96b9-d2d464bcbcc0.png)

`
### WordPress Environment ###

Home URL: -
Site URL: -
WC Version: 3.1.0
Log Directory Writable: ✔
WP Version: 4.8
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: pt_BR

### Server Environment ###

Server Info: Apache
PHP Version: 5.6.30
PHP Post Max Size: 64 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.19.7
NSS/3.27.1

SUHOSIN Installed: –
MySQL Version: ❌ 5.5.51 - Nós recomendamos utilizar pelo menos a versão 5.6 do MySQL. Mais informações: Requerimentos do WordPress
Max Upload Size: 64 MB
Default Timezone is UTC: ❌ O fuso horário padrão é America/Sao_Paulo - deve ser UTC
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.1.0
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Security ###

Secure connection (HTTPS): ❌Sua loja não está usando HTTPS. Saiba mais sobre HTTPS e certificados SSL.
Hide errors from visitors: ✔

### Active Plugins (12) ###

Fix Image Orientation: por Tim Carr – 1.0.0
CMB2: por WebDevStudios – 2.2.4
noSebbo: por WC Marketplace
The Grey Parrots – 2.7.3

Tradução (Loco): por Tim Whitlock – 2.0.14
Ninja Forms: por The WP Ninjas – 3.1.6
Slider Revolution: por ThemePunch – 5.4.3.1
Super Socializer: por Team Heateor – 7.9.2
WC Password Strength Settings: por Daniel Santoro – 1.1.0
Woo - Autocomplete Orders after x days: por Renato Nagliati – 0.1
WooCommerce Extra Checkout Fields for Brazil: por Claudio Sanches – 3.6.0
WooCommerce: por Automattic – 3.1.0
WooDiscuz - WooCommerce Comments: por gVectors Team (A. Chakhoyan
G. Zakaryan
H. Martirosyan) – 2.0.12


### Settings ###

API Enabled: ✔
Force SSL: –
Currency: BRL (R$)
Currency Position: left_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Base da loja: #4 - /loja/
Carrinho: #5 - /carrinho/
Finalizar compra: #6 - /finalizar-compra/
Minha conta: #7 - /minha-conta/


Thanks,
Renato
